### PR TITLE
Fix for loading atlas attachments

### LINKED ISSUE #140: Crash when running animation from a timer and a coroutine
If i run animation from coroutine, while other animation already playing,  game window some times crashes

```
---@param character_id hash
---@param anim_id string|hash
---@param blend_duration number?
function M.play_anim_in_coroutine(character_id, anim_id, blend_duration, playback)
    local character_view_id = maputil.resolve_character(character_id, true)
    local character_spine_url = msg.url(nil, character_view_id,"spinemodel")
    spine.play_anim(character_spine_url, anim_id, playback or go.PLAYBACK_ONCE_FORWARD, { blend_duration = blend_duration or .1 }, function(_, message_id)
        if message_id == const.SPINE_ANIMATION_DONE then
         ...
        end
    end)
end
```
   
On windows

```
ERROR:CRASH: CALL STACK:
ERROR:CRASH:  0 0x7FF6516E2420 lua_pushinteger <unknown>:0
ERROR:CRASH:  1 0x7FF651714820 dmScript::GetInstanceContextTable D:\a\defold\defold\engine\script\src\script.cpp:986
ERROR:CRASH:  2 0x7FF65170F630 dmScript::UnrefInInstance D:\a\defold\defold\engine\script\src\script.cpp:1291
ERROR:CRASH:  3 0x7FF651633330 dmSpine::PlayAnimation /tmp/job12849756285391918148/upload/defold-spine/src/comp_spine_model.cpp:349
ERROR:CRASH:  4 0x7FF651635430 dmSpine::CompSpineModelPlayAnimation /tmp/job12849756285391918148/upload/defold-spine/src/comp_spine_model.cpp:956
ERROR:CRASH:  5 0x7FF651652700 dmSpine::SpineComp_PlayAnim /tmp/job12849756285391918148/upload/defold-spine/src/script_spine.cpp:330
ERROR:CRASH:  6 0x7FF651655E1E lj_BC_FUNCC <unknown>:0
ERROR:CRASH:  7 0x7FF6516E3050 lua_pcall <unknown>:0
ERROR:CRASH:  8 0x7FF651714BE0 dmScript::PCallInternal D:\a\defold\defold\engine\script\src\script.cpp:1387
ERROR:CRASH:  9 0x7FF651711970 dmScript::InvokeCallback D:\a\defold\defold\engine\script\src\script.cpp:1779
ERROR:CRASH: 10 0x7FF651746D90 dmGameObject::LuaAnimationStopped D:\a\defold\defold\engine\gameobject\src\gameobject\gameobject_script.cpp:1555
ERROR:CRASH: 11 0x7FF6517603D0 dmGameObject::CompAnimUpdate D:\a\defold\defold\engine\gameobject\src\gameobject\comp_anim.cpp:342
ERROR:CRASH: 12 0x7FF651740670 dmGameObject::Update D:\a\defold\defold\engine\gameobject\src\gameobject\gameobject.cpp:2577
ERROR:CRASH: 13 0x7FF6517F9F80 dmGameSystem::CompCollectionProxyUpdate D:\a\defold\defold\engine\gamesys\src\gamesys\components\comp_collection_proxy.cpp:299
ERROR:CRASH: 14 0x7FF651740670 dmGameObject::Update D:\a\defold\defold\engine\gameobject\src\gameobject\gameobject.cpp:2577
ERROR:CRASH: 15 0x7FF6517F9F80 dmGameSystem::CompCollectionProxyUpdate D:\a\defold\defold\engine\gamesys\src\gamesys\components\comp_collection_proxy.cpp:299
ERROR:CRASH: 16 0x7FF651740670 dmGameObject::Update D:\a\defold\defold\engine\gameobject\src\gameobject\gameobject.cpp:2577
ERROR:CRASH: 17 0x7FF6517A6320 dmEngine::StepFrame D:\a\defold\defold\engine\engine\src\engine.cpp:1577
ERROR:CRASH: 18 0x7FF6517A1510 dmEngineUpdate D:\a\defold\defold\engine\engine\src\engine.cpp:2105
ERROR:CRASH: 19 0x7FF6517A6D50 dmEngine::RunLoop D:\a\defold\defold\engine\engine\src\engine_loop.cpp:83
ERROR:CRASH: 20 0x7FF6516D9800 engine_main D:\a\defold\defold\engine\engine\src\engine_main.cpp:148
ERROR:CRASH: 21 0x7FF651A2A2E4 __scrt_common_main_seh d:\agent\_work\4\s\src\vctools\crt\vcstartup\src\startup\exe_common.inl:288
ERROR:CRASH: 22 0x7FFA9DE12690 BaseThreadInitThunk <unknown>:0
ERROR:CRASH: 23 0x7FFA9F9AA9D0 RtlUserThreadStart <unknown>:0
```

On mac

```
ERROR:SCRIPT: ../src/script.cpp:1286: LuaStackCheck: m_Diff >= -m_Top == false (m_Diff: 0, m_Top: -1676072)
Assertion failed: (m_Diff >= -m_Top), function LuaStackCheck, file script.cpp, line 1468.
INFO:CRASH: Successfully wrote Crashdump to file: /Users/salavat/Library/Application Support/Defold/_crash
ERROR:CRASH: CALL STACK:

# 0 pc     0x2f2e94 dmengine _ZN7dmCrashL7HandlerEiP9__siginfoPv+36
# 1 pc     0x2f05ed libxpc.dylib _sigtramp+29
# 2 pc     0x28b1f3 libxpc.dylib __pthread_kill+11
# 3 pc     0x2c2ee6 libxpc.dylib pthread_kill+263
# 4 pc     0x1e9b45 libxpc.dylib abort+123
# 5 pc     0x1e8e5e libxpc.dylib err+0
# 6 pc     0x3e61a2 dmengine _ZN8dmScript15UnrefInInstanceEP9lua_Statei.cold.1+82
# 7 pc     0x1b97a3 dmengine _ZN8dmScript15UnrefInInstanceEP9lua_Statei+115
# 8 pc      0x372ec dmengine _ZN7dmSpineL13PlayAnimationEPNS_19SpineModelComponentEyN12dmGameObject8PlaybackEfffi+588
# 9 pc      0x38c0d dmengine _ZN7dmSpine27CompSpineModelPlayAnimationEPNS_19SpineModelComponentEPN15dmGameSystemDDF18SpinePlayAnimationEPN9dmMessage3URLEiP9lua_State+61
#10 pc      0x3df09 dmengine _ZN7dmSpineL18SpineComp_PlayAnimEP9lua_State+777
#11 pc     0x1d8776 dmengine lj_BC_FUNCC+68
#12 pc     0x1d95ac dmengine lj_ff_coroutine_resume+176
#13 pc     0x1f1d2b dmengine lua_pcall+155
#14 pc     0x1b98c1 dmengine _ZN8dmScriptL13PCallInternalEP9lua_Stateiii+81
#15 pc      0x92c4d dmengine _ZN12dmGameObject9RunScriptEP9lua_StatePNS_6ScriptENS_14ScriptFunctionEPNS_14ScriptInstanceERKNS_15RunScriptParamsE+365
#16 pc      0x92f5c dmengine _ZN12dmGameObjectL24CompScriptUpdateInternalERKNS_22ComponentsUpdateParamsENS_14ScriptFunctionERNS_22ComponentsUpdateResultE+156
#17 pc      0x9d887 dmengine _ZN12dmGameObject6UpdateEPNS_16CollectionHandleEPKNS_13UpdateContextE+503
#18 pc      0xbb8c2 dmengine _ZN12dmGameSystem25CompCollectionProxyUpdateERKN12dmGameObject22ComponentsUpdateParamsERNS0_22ComponentsUpdateResultE+610
#19 pc      0x9d887 dmengine _ZN12dmGameObject6UpdateEPNS_16CollectionHandleEPKNS_13UpdateContextE+503
#20 pc      0xbb8c2 dmengine _ZN12dmGameSystem25CompCollectionProxyUpdateERKN12dmGameObject22ComponentsUpdateParamsERNS0_22ComponentsUpdateResultE+610
#21 pc      0x9d887 dmengine _ZN12dmGameObject6UpdateEPNS_16CollectionHandleEPKNS_13UpdateContextE+503
#22 pc      0x42f85 dmengine _ZN8dmEngine4StepEPNS_6EngineE+1077
#23 pc      0x4361a dmengine _Z14dmEngineUpdatePN8dmEngine6EngineE+26
#24 pc      0x4414d dmengine _ZN8dmEngine7RunLoopEPKNS_13RunLoopParamsE+109
#25 pc      0x440b2 dmengine _Z11engine_mainiPPc+98

```

### PR DESCRIPTION
**Pull request by JCash**
Fixes #140

### CODE REVIEW

File: defold-spine/commonsrc/spine_loader.cpp [hunk 1]
```diff
@@ -212,8 +212,12 @@ namespace dmSpine
     static bool loadSequence(dmHashTable64<uint32_t>* name_to_index, spAtlasRegion* atlasRegions, const char *basePath, spSequence *sequence, spAtlasRegion* default_region) {
         bool is_atlas_available = name_to_index != 0;
 
+        // The C runtime seems a bit broken,
+        // and doesn't account for number of digits required regions->size
+        // So, instead, we simply add some extra room for X digits
+        uint32_t region_digits = 8;
```
Comment (JCash) [inline defold-spine/commonsrc/spine_loader.cpp:218]: Notable that they haven't encountered this themselves?

**Review by AGulev — APPROVED** [review]
oooh
great find!

**Comment by AGulev** [comment]
@JCash is it fix for https://github.com/defold/extension-spine/issues/140 ?

**Comment by JCash** [comment]
It's a good question. 🤔 
The only error I got was due to this memory overwrite (caught with asan).
(I didn't look too closely at the ticket's callstack)

While the overwrite happened during the loading, it guess might still affect later code (e.g. the callbacks).

In any case, I couldn't reproduce the issue descibed in the ticket.

**Pull request merged**